frontend/src/stores/project.js

```diff
@@ -0,0 +1,175 @@
+                );
+
+                return result;
+            };
+        },
+
+        jsonForCompile() {
+            return {
+                code: this.debugCode,
+                language: this.language,
+                inputs: this.testData.map((testCase) => testCase.input),
+            };
+        },
+    },
+
+    actions: {
+        /* Setters */
+        setCode(newValue) {
+            this.code = newValue;
+        },
+        setBreakpoints(newValue) {
+            this.breakpoints = newValue;
+        },
+        setIsRunning(newValue) {
+            this.isRunning = newValue;
+        },
+        setWaitingForCompile(newValue) {
+            this.waitingForCompile = newValue;
+        },
+        switchCurrentTestCase(index) {
+            this.currentTestCaseId = index;
+        },
+        switchCurrentFrame(index) {
+            this.currentFrameId = index;
+        },
+
+        /* Breakpoints */
+        toggleBreakpoint(id) {
+            this.breakpoints.toggleElement({ id: id });
+        },
+
+        /* TestData */
+        addTestCase() {
+            this.testData.addElement({ input: "" });
+        },
+        deleteTestCase(id) {
+            this.testData.deleteById(id);
+        },
+        updateCurrentTestCaseInput(newValue) {
+            this.testData.findById(this.currentTestCaseId).input = newValue;
+        },
+
+        /* SceneObjects */
+        addSceneObject(sceneObject) {
+            this.sceneObjects.addElement(sceneObject);
+        },
+        deleteSceneObject(index) {
+            this.sceneObjects.deleteById(index);
+        },
+        updateSceneObjectPosition(payload) {
+            this.sceneObjects.findById(payload.id).position = payload.position;
+        },
+
+        /* Variables */
+        updateVariable(payload) {
+            const { id, variable } = payload;
+            variable.id = variable.name = this.code.substring(variable.start, variable.end);
+            this.sceneObjects
+                .flatMap((sceneObject) => [sceneObject, ...sceneObject.subobjects])
+                .find((sceneObject) => sceneObject.variable?.id === id).variable = variable;
+        },
+        deleteVariable(id) {
+            this.sceneObjects
+                .flatMap((sceneObject) => [sceneObject, ...sceneObject.subobjects])
+                .find((sceneObject) => sceneObject.variable?.id === id).variable = null;
+        },
+
+        /* Logic */
+        loadProject(projectId) {
+            sendRequest("/project/find/" + projectId, null, "GET").then((responseData) => {
+                const project = responseData;
+                ["_id", "title", "code", "breakpoints", "language", "testData", "sceneObjects", "title"].forEach(
+                    (property) => (this[property] = project[property])
+                );
+                this.currentTestCaseId = project.testData.firstId();
+            });
+        },
+
+        saveProject({ override, title }) {
```

Comment:
Domyślam się, że Pinia umożliwia tworzenie funkcji z kilkoma argumentami? W vueX się tego nie dało zrobić, trzeba było to właśnie złączyć w jeden obiekt. Jeśli da się to, proponuję rozdzielić to na 2 argumenty (i poprawić w miejscu wywołania tej funkcji).